src/apps/subjects/crud/subject.py

```diff
@@ -48,6 +48,13 @@ async def get_by_ids(self, ids: list[uuid.UUID]) -> list[SubjectSchema]:
         res = await self._execute(query)
         return res.scalars().all()
 
+    async def get_all_by_ids(self, ids: list[uuid.UUID]) -> list[SubjectSchema]:
+        """Get all subjects by ids, including soft deleted subjects"""
+        query: Query = select(SubjectSchema)
+        query = query.where(SubjectSchema.id.in_(ids))
+        res = await self._execute(query)
+        return res.scalars().all()
+
```

Comment:
The code seems pretty similar as the existing `get_by_ids`. Is it possible to replace and combine both versions into one by using an optional parameter to include `soft_exists`?
```suggestion
    async def get_by_ids(self, ids: list[uuid.UUID], include_deleted: bool = False) -> list[SubjectSchema]:
        query: Query = select(SubjectSchema)
        if include_deleted:
            query = query.where(SubjectSchema.id.in_(ids))
        else:
            query = query.where(SubjectSchema.id.in_(ids), SubjectSchema.soft_exists())
        res = await self._execute(query)
        return res.scalars().all()

```

Reply:
yes, it can be done this way, put I personally do not like boolean flag parameters to change function behavior, it is not wrong, but I use more functional way to write code with less conditionals. I can change it if you want

Reply:
we have similar code design in get_applet_answers with include_assessment parameter, so to keep code pattern we should go with you suggestion

Reply:
Updated!